apps/web/hooks/useLinkResolver/useLinkResolver.tsx

```diff
@@ -178,9 +193,18 @@ export const typeResolver = (
 
       // convert the route template string into a regex query
       const regex = convertToRegex(routeTemplate)
-      // if this path matches query return route info else continue
-      if (path?.match(regex)) {
-        return { type, locale } as TypeResolverResponse
+
+      // if the path starts with the routeTemplate string or matches dynamic route regex we have found the type
+      if (
+        (!skipDynamic && path?.match(regex)) ||
+        (skipDynamic && path?.startsWith(routeTemplate))
+      ) {
+        console.log('Found type!', type, locale)
+        return {
+          slug: extractSlugsByRouteTemplate(path, routeTemplate),
+          type,
+          locale,
+        } as TypeResolverResponse
```

Comment:
is this method tested? do you need to add more here or update the tests to consider this case?

Reply:
Yes, this is tested via typeResolver test but will add tests specific to this